Replicator/DBAccess.hh

```diff
@@ -73,6 +73,9 @@ namespace litecore { namespace repl {
 
         /** Returns the remote ancestor revision ID of a document. */
         alloc_slice getDocRemoteAncestor(C4Document *doc);
+        
+        /** Returns the document enumerator for all unresolved docs present in the DB */
+        C4DocEnumerator* unresolvedDocsEnumerator(C4Error *outError);
```

Comment:
returning the docEnumerator here can avoid help avoid unwanted conversion to docs and then to RevToInsert.